### PR TITLE
Fix test_run_check unit test for Linux and Windows

### LINKED ISSUE #6338: Fim registry improvement unit tests, test_run_check file remake
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.0.0 | Syscheck | Agent | Sources | Windows |

## test_run_check

After the changes of the Fim registry improvement, it is necessary to modify some of the tests related to the following functions:
- fim_send_sync_msg
- fim_link_update
- fim_link_check_delete
- fim_delete_realtime_watches
- fim_link_delete_range

### PR DESCRIPTION
**Pull request by antoniomanuelfr**
|Related issue|
|---|
|#6338|

## Description

After the development of #5576, the unit tests need to be adapted. This PR aims to adapt the tests in the file `test_run_check.c`
Closes #6338 
## Linux tests
```
[==========] Running 21 test(s).
[ RUN      ] test_fim_whodata_initialize
[       OK ] test_fim_whodata_initialize
[ RUN      ] test_log_realtime_status
[       OK ] test_log_realtime_status
[ RUN      ] test_fim_send_msg
[       OK ] test_fim_send_msg
[ RUN      ] test_fim_send_msg_retry
[       OK ] test_fim_send_msg_retry
[ RUN      ] test_fim_send_msg_retry_error
[       OK ] test_fim_send_msg_retry_error
[ RUN      ] test_fim_send_sync_msg_10_eps
[       OK ] test_fim_send_sync_msg_10_eps
[ RUN      ] test_fim_send_sync_msg_0_eps
[       OK ] test_fim_send_sync_msg_0_eps
[ RUN      ] test_send_syscheck_msg_10_eps
[       OK ] test_send_syscheck_msg_10_eps
[ RUN      ] test_send_syscheck_msg_0_eps
[       OK ] test_send_syscheck_msg_0_eps
[ RUN      ] test_fim_send_scan_info
[       OK ] test_fim_send_scan_info
[ RUN      ] test_fim_link_update
[       OK ] test_fim_link_update
[ RUN      ] test_fim_link_update_already_added
[       OK ] test_fim_link_update_already_added
[ RUN      ] test_fim_link_check_delete
[       OK ] test_fim_link_check_delete
[ RUN      ] test_fim_link_check_delete_lstat_error
[       OK ] test_fim_link_check_delete_lstat_error
[ RUN      ] test_fim_link_check_delete_noentry_error
[       OK ] test_fim_link_check_delete_noentry_error
[ RUN      ] test_fim_delete_realtime_watches
[       OK ] test_fim_delete_realtime_watches
[ RUN      ] test_fim_link_delete_range
[       OK ] test_fim_link_delete_range
[ RUN      ] test_fim_link_delete_range_error
[       OK ] test_fim_link_delete_range_error
[ RUN      ] test_fim_link_silent_scan
[       OK ] test_fim_link_silent_scan
[ RUN      ] test_fim_link_reload_broken_link_already_monitored
[       OK ] test_fim_link_reload_broken_link_already_monitored
[ RUN      ] test_fim_link_reload_broken_link_reload_broken
[       OK ] test_fim_link_reload_broken_link_reload_broken
[==========] 21 test(s) run.
[  PASSED  ] 21 test(s).
```
## Windows tests
```
wine test_run_check.exe 
[==========] tests: Running 17 test(s).
[ RUN      ] test_fim_whodata_initialize
[       OK ] test_fim_whodata_initialize
[ RUN      ] test_set_priority_windows_thread_highest
[       OK ] test_set_priority_windows_thread_highest
[ RUN      ] test_set_priority_windows_thread_above_normal
[       OK ] test_set_priority_windows_thread_above_normal
[ RUN      ] test_set_priority_windows_thread_normal
[       OK ] test_set_priority_windows_thread_normal
[ RUN      ] test_set_priority_windows_thread_below_normal
[       OK ] test_set_priority_windows_thread_below_normal
[ RUN      ] test_set_priority_windows_thread_lowest
[       OK ] test_set_priority_windows_thread_lowest
[ RUN      ] test_set_priority_windows_thread_idle
[       OK ] test_set_priority_windows_thread_idle
[ RUN      ] test_set_priority_windows_thread_error
[       OK ] test_set_priority_windows_thread_error
[ RUN      ] test_log_realtime_status
[       OK ] test_log_realtime_status
[ RUN      ] test_fim_send_msg
[       OK ] test_fim_send_msg
[ RUN      ] test_fim_send_msg_retry
[       OK ] test_fim_send_msg_retry
[ RUN      ] test_fim_send_msg_retry_error
[       OK ] test_fim_send_msg_retry_error
[ RUN      ] test_fim_send_sync_msg_10_eps
[       OK ] test_fim_send_sync_msg_10_eps
[ RUN      ] test_fim_send_sync_msg_0_eps
[       OK ] test_fim_send_sync_msg_0_eps
[ RUN      ] test_send_syscheck_msg_10_eps
[       OK ] test_send_syscheck_msg_10_eps
[ RUN      ] test_send_syscheck_msg_0_eps
[       OK ] test_send_syscheck_msg_0_eps
[ RUN      ] test_fim_send_scan_info
[       OK ] test_fim_send_scan_info
[==========] tests: 17 test(s) run.
[  PASSED  ] 17 test(s).

```
```
wine test_run_check_event.exe 
[==========] eventchannel_tests: Running 3 test(s).
[ RUN      ] test_set_whodata_mode_changes
[       OK ] test_set_whodata_mode_changes
[ RUN      ] test_fim_whodata_initialize_eventchannel
[       OK ] test_fim_whodata_initialize_eventchannel
[ RUN      ] test_fim_whodata_initialize_fail_set_policies
[       OK ] test_fim_whodata_initialize_fail_set_policies
[==========] eventchannel_tests: 3 test(s) run.
[  PASSED  ] 3 test(s).

```

### CODE REVIEW

File: src/unit_tests/syscheckd/test_run_check.c [hunk 1]
```diff
@@ -154,6 +158,15 @@ static int teardown_tmp_file(void **state) {
 }
 #endif
 
+/**
+ * @brief This function loads expect and will_return calls for the function send_sync_msg
+*/
+static void expect_w_send_sync_msg(const char *msg, const char *locmsg, int ret, int location) {
```
Comment (jesusjimsa) [inline src/unit_tests/syscheckd/test_run_check.c:164]: Move this function to the `mq_op_wrappers` file.

Comment (jesusjimsa) [inline src/unit_tests/syscheckd/test_run_check.c:164]: Move this function to the `mq_op_wrappers` file.

File: src/unit_tests/syscheckd/test_run_check.c [hunk 2]
```diff
@@ -500,20 +448,18 @@ void test_fim_whodata_initialize_eventchannel(void **state) {
 }
 #endif  // WIN_WHODATA
 #endif
+
 void test_fim_send_sync_msg_10_eps(void ** state) {
     (void) state;
     syscheck.sync_max_eps = 10;
+    char location[10] = "fim_file\0";
```
Comment (jesusjimsa) [inline src/unit_tests/syscheckd/test_run_check.c:455]: Is the `\0` character necessary? If not, please delete it.

File: src/unit_tests/syscheckd/test_run_check.c [hunk 3]
```diff
@@ -524,26 +470,18 @@ void test_fim_send_sync_msg_10_eps(void ** state) {
...
-    fim_send_sync_msg("fim_file", "");
+    expect_w_send_sync_msg("", location, 0, DBSYNC_MQ);
+    fim_send_sync_msg( location, "");
 }
 
 void test_fim_send_sync_msg_0_eps(void ** state) {
     (void) state;
     syscheck.sync_max_eps = 0;
-
+    char location[10] = "fim_file\0";
```
Comment (jesusjimsa) [inline src/unit_tests/syscheckd/test_run_check.c:480]: The same as above, is the null character necessary?

File: src/unit_tests/wrappers/wazuh/shared/mq_op_wrappers.c [hunk 4]
```diff
@@ -25,3 +25,12 @@ int __wrap_StartMQ(const char *path, short int type,__attribute__((unused)) shor
     check_expected(type);
     return mock();
 }
+
+/**
+ * @brief This function load the expect and will_return calls for the function StartMQ
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/shared/mq_op_wrappers.c:30]: This function **loads**. Also, this doxygen comment should be in the _.h_ file instead.

File: src/unit_tests/wrappers/wazuh/syscheckd/create_db_wrappers.c [hunk 5]
```diff
@@ -64,3 +64,21 @@ int __wrap_fim_whodata_event(whodata_evt * w_evt)
 #endif
     return 1;
 }
+
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of fim_checker
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/create_db_wrappers.c:69]: This function **loads**. Also, this doxygen comment should be in the _.h_ file instead.

File: src/unit_tests/wrappers/wazuh/syscheckd/create_db_wrappers.c [hunk 6]
```diff
@@ -64,3 +64,21 @@ int __wrap_fim_whodata_event(whodata_evt * w_evt)
...
+*/
+void expect_fim_checker_call(const char *path, int w_evt, int report) {
+    expect_string(__wrap_fim_checker, path, path);
+    expect_value(__wrap_fim_checker, w_evt, w_evt);
+    expect_value(__wrap_fim_checker, report, report);
+}
+
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of fim_configuration_directory
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/create_db_wrappers.c:78]: This function **loads**. Also, this doxygen comment should be in the _.h_ file instead.

File: src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.c [hunk 7]
```diff
@@ -190,3 +192,33 @@ int __wrap_fim_db_sync_path_range(fdb_t *fim_sql,
 
     return mock();
 }
+
+/* Auxiliar functions */
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of fim_db_get_path_range
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.c:198]: This function **loads**. Also, this doxygen comment should be in the _.h_ file instead.

File: src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.c [hunk 8]
```diff
@@ -190,3 +192,33 @@ int __wrap_fim_db_sync_path_range(fdb_t *fim_sql,
...
+    expect_value(__wrap_fim_db_get_path_range, type, FIM_TYPE_FILE);
+    expect_string(__wrap_fim_db_get_path_range, start, start_str);
+    expect_string(__wrap_fim_db_get_path_range, top, top_str);
+    expect_value(__wrap_fim_db_get_path_range, storage, storage);
+    will_return(__wrap_fim_db_get_path_range, tmp_file);
+    will_return(__wrap_fim_db_get_path_range, ret);
+}
+
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of fim_db_delete_range
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.c:217]: This function **loads**. Also, this doxygen comment should be in the _.h_ file instead.

File: src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.c [hunk 9]
```diff
@@ -190,3 +192,33 @@ int __wrap_fim_db_sync_path_range(fdb_t *fim_sql,
...
+/* Auxiliar functions */
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of fim_db_get_path_range
+*/
+void expect_fim_db_get_path_range_call(const fdb_t *db,
+                                           const char *start_str,
+                                           const char *top_str,
+                                           int storage,
+                                           fim_tmp_file *tmp_file,
+                                           int ret) {
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.c:205]: These lines are over-indented.
```C
void expect_fim_db_get_path_range_call(const fdb_t *db,
                                       const char *start_str,
                                       const char *top_str,
                                       int storage,
                                       fim_tmp_file *tmp_file,
                                       int ret) {
```

File: src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.h [hunk 10]
```diff
@@ -92,4 +93,17 @@ int __wrap_fim_db_sync_path_range(fdb_t *fim_sql,
                                   fim_tmp_file *file,
                                   int storage);
 
+/* Auxiliar functions */
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of fim_db_get_path_range
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.h:98]: This function **loads**.

File: src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.h [hunk 11]
```diff
@@ -92,4 +93,17 @@ int __wrap_fim_db_sync_path_range(fdb_t *fim_sql,
...
+*/
+void expect_fim_db_get_path_range_call(const fdb_t *db,
+                                    const char *start_str,
+                                    const char *top_str,
+                                    int storage,
+                                    fim_tmp_file *tmp_file,
+                                    int ret);
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.h:105]: These lines are under-indented.
```C
void expect_fim_db_get_path_range_call(const fdb_t *db,
                                       const char *start_str,
                                       const char *top_str,
                                       int storage,
                                       fim_tmp_file *tmp_file,
                                       int ret);
```

File: src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.h [hunk 12]
```diff
@@ -92,4 +93,17 @@ int __wrap_fim_db_sync_path_range(fdb_t *fim_sql,
...
+
+void expect_fim_db_delete_range_call(const fdb_t *db, int storage, const fim_tmp_file *file, int ret);
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/wazuh/syscheckd/fim_db_wrappers.h:107]: This function doesn't have doxygen information as the other ones you added.

File: src/unit_tests/wrappers/windows/processthreadsapi_wrappers.c [hunk 13]
```diff
@@ -43,3 +43,12 @@ BOOL wrap_OpenProcessToken(__UNUSED_PARAM (HANDLE  ProcessHandle),
     *TokenHandle = mock_type(HANDLE);
     return mock();
 }
+
+/**
+ * @brief This function load the expect and will_return calls for the wrapper of SetThreadPriority
```
Comment (jesusjimsa) [inline src/unit_tests/wrappers/windows/processthreadsapi_wrappers.c:48]: This function **loads**. Also, this doxygen comment should be in the _.h_ file instead.

**Review by jesusjimsa — CHANGES_REQUESTED** [review]

**Review by jesusjimsa — APPROVED** [review]
LGTM!

**Review by Molter73 — APPROVED** [review]
LGTM!

**Pull request merged**